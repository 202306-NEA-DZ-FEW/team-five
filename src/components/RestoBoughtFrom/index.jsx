import { useTranslation } from "next-i18next";
import React from "react";

const BoughtFrom = (props) => {
    const { selectedRestaurants } = props;
    const { t } = useTranslation("payment");
    if (selectedRestaurants.length === 0) {
        return null;
    }
    return (
        <div className='restaurant-section lg:flex lg:flex-row flex flex-col items-center'>
            <h2 className='text-2xl font-semibold mb-4'>
                {t("payment.title")}
            </h2>
            <ul className='retaurant-list list-none p-0 flex flex-wrap gap-10'>
                {selectedRestaurants.map((restaurant, index) => (
                    <li key={index} className='retaurant-item lg:w-1/3 w-full'>
                        {restaurant}
                    </li>
                ))}
            </ul>
        </div>
    );
};

export default BoughtFrom;
