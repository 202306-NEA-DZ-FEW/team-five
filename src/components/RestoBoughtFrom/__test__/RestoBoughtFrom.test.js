import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import RestoBoughtFrom from "..";

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("Resto Bought From Component", () => {
    it("should render correctly", () => {
        const selectedRestaurants = [{ id: 1, name: "Restaurant A" }];

        const component = renderer.create(
            <Provider store={store}>
                <RestoBoughtFrom
                    selectedRestaurants={selectedRestaurants.map(
                        (restaurant) => restaurant.name
                    )}
                />
            </Provider>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
