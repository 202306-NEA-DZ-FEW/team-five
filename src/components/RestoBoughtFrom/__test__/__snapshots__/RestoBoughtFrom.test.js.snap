// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resto Bought From Component should render correctly 1`] = `
<div
  className="restaurant-section lg:flex lg:flex-row flex flex-col items-center"
>
  <h2
    className="text-2xl font-semibold mb-4"
  >
    payment.title
  </h2>
  <ul
    className="retaurant-list list-none p-0 flex flex-wrap gap-10"
  >
    <li
      className="retaurant-item lg:w-1/3 w-full"
    >
      Restaurant A
    </li>
  </ul>
</div>
`;
